app/lib/pdf_filler/nj1040_pdf.rb

```diff
@@ -20,6 +20,14 @@ def hash_for_pdf
       mfj_spouse_ssn = get_mfj_spouse_ssn
       mfs_spouse_ssn = get_mfs_spouse_ssn
       answers = {
+        # header
+        'Your Social Security Number': taxpayer_ssn.to_s,
```

Comment:
is this header getting reflected in the XML as well? @jachan

Reply:
Nope! Just in the PDF, it reuses existing XML data and puts it in the header of the PDF on pages 2-4.

Reply:
Seeing that the Primary Name/SSN appears to be stored as part of the return header: https://github.com/codeforamerica/vita-min/blob/main/app/lib/submission_builder/return_header.rb#L20. Testing shows that the XML doc generated by the application stores the name under:

```
<ReturnHeaderState>
  <Filer>
    <Primary>
      <TaxpayerName>
        <First Name>
        <Middle Initial>
        <Last Name>
        <Name Suffix>
```

To combine these fields for the PDF, we use the standard NJ format, which differs based on information provided: https://github.com/codeforamerica/vita-min/blob/main/spec/lib/pdf_filler/nj1040_pdf_spec.rb#L186

Reply:
@jachan Yes nice! Since its part of the XML in the return header XML I think it makes more sense to grab the fields from there like so `@xml_document.at('Primary TaxpayerSSN')&.text` or `@xml_document.at('Secondary TaxpayerSSN')&.text` (see `app/lib/pdf_filler/az140_pdf.rb`)

Reply:
Yep! That's already getting done, `taxpayer_ssn = get_taxpayer_ssn` which sources from the XML itself https://github.com/codeforamerica/vita-min/blob/3f5ff0e4eab5c7ebd81a87deaf1bba9ea14d665a/app/lib/pdf_filler/nj1040_pdf.rb#L146